src/compiler/crystal/tools/doc/method.cr

```diff
@@ -57,15 +57,11 @@ class Crystal::Doc::Method
   private def compute_doc_info : DocInfo?
     def_doc = @def.doc
     if def_doc
-      has_inherit = def_doc =~ /^\s*:inherit:\s*$/m
-      if has_inherit
-        ancestor_info = self.ancestor_doc_info
-        if ancestor_info
-          def_doc = def_doc.gsub(/^[ \t]*:inherit:[ \t]*$/m, ancestor_info.doc.not_nil!)
-          return DocInfo.new(def_doc, nil)
-        end
+      # TODO: warn about `:inherit:` not finding an ancestor
+      inherit_def_doc = def_doc.gsub(/^[ \t]*:inherit:[ \t]*$/m) { ancestor_doc_info.try(&.doc) || break }
```

Comment:
This would recompute `#ancestor_doc_info` on every iteration. Normally docs shouldn't contain more than one `:inherit:` but it doesn't hurt to cache that in a local variable

Reply:
Now it's still being recomputed if `self.inherit_def_doc` returns `nil`. This double meaning of `nil` is also bad for understanding the code. I'd just drop the caching.

Reply:
> Now it's still being recomputed if `self.inherit_def_doc` returns `nil`. This double meaning of `nil` is also bad for understanding the code. I'd just drop the caching (revert that commit).

(this issue still stands; reformatting didn't change the behavior)

Reply:
Ah sorry, I missed your comment here @oprypin 

I suppose you mean `self.ancestor_doc_info`?
That doesn't get recomputed because when it returns nil, the block breaks and there's not repeat iteration.